### PR TITLE
Added return value in __Pyx_UnicodeContainsUCS4

### PR DESCRIPTION
**Pull request by da-woods**
https://github.com/cython/cython/issues/3925

This is a corner-case mostly to silence C warnings. It can happen
with strings created manually (i.e. not by Cython) using the deprecated
C-API on Python 3.9+

I haven't added a test since it seems quite hard to construct a
string that fails it

### CODE REVIEW

**Comment by da-woods** [comment]
The alternative would be to fall back to `PyUnicode_FindChar` or `PyUnicode_Contains` on the understanding that this would be a rarely used slow path. Both of these would require Python error handling (in a function that isn't supposed to return exceptions)

File: Cython/Utility/StringTools.c [hunk 1]
```diff
@@ -176,6 +176,15 @@ static CYTHON_INLINE int __Pyx_UnicodeContainsUCS4(PyObject* unicode, Py_UCS4 ch
 
     }
 #endif
+    /* This should only happen if the user has manually created a non-PEP393 string
+     * (i.e. it's very unlikely)
```
Comment (scoder) [inline Cython/Utility/StringTools.c:180]: wchar strings are actually a common thing on Windows, where that's kind-of the native system Unicode string type. Windows file names fall into that category, for example.

**Comment by scoder** [comment]
The right fallback is probably what already follows just below for the pre-3.9 case, except that we'd have to use `wchar_t` instead of `Py_UNICODE`.

**Comment by da-woods** [comment]
> The right fallback is probably what already follows just below for the pre-3.9 case, except that we'd have to use `wchar_t` instead of `Py_UNICODE`.

That still requires `PyUnicode_AS_UNICODE` - there doesn't seem to be a way to access the data as `wchar_t` instead of `Py_UNICODE`.

That suggests to me that the right thing to do is revert https://github.com/cython/cython/commit/336363893b6cc02f6c1d0f26102ea2501784f7d7 but wrap the whole thing in

```
#pragma GCC diagnostic push  // works for clang too
#prgama GCC diagnostic ignored "-Wdeprecated"

// code...

#pragma GCC diagnostic pop
```

(just to make the warnings go away.) Possibly with something similar for MSVC too.

**Comment by scoder** [comment]
Actually, `WCHAR_KIND` is [already scheduled for removal](https://www.python.org/dev/peps/pep-0623/) in Py3.12 and `PyUnicode_IS_READY()` as well. I'll see if I can come up with something reasonable here.

**Comment by scoder** [comment]
See #4135 for an attempt to make this more future proof.

**Comment by scoder** [comment]
Superseded by #4135.
